### PR TITLE
Introduce event specific processors

### PR DESCRIPTION
**Pull request by dturgumbaev**
- Add event specific attributes
- refactor FPTI event input mapping: the event to FPTInput is done in one place. This constructs FPTIInput from device information, user identity and event to FPTI mapping
- Deprecate productView

### CODE REVIEW

File: src/lib/shopping-fpti/event-handlers.js [hunk 1]
```diff
@@ -0,0 +1,49 @@
...
+      return {
+        es: 'visitorInfoFlowStarted',
+        fltp: 'store-cash',
+        offer_id: offerId
+      };
+    }
+    return {};
+  }
+
+  function enrichPurchaseEvent() : Object {
```
Comment (batiskafff) [inline src/lib/shopping-fpti/event-handlers.js:18]: Why does it named **enrich**PurchaseEvent ?

File: src/types/fpti.js [hunk 2]
```diff
@@ -53,7 +53,13 @@ export type FptiInput = {|
   // SDK generated user ID
   shopperId? : string,
   // Merchant provided user ID
-  merchantProvidedUserId? : string
+  merchantProvidedUserId? : string,
+  
+  es? : string,
+
+  fltp? : string,
```
Comment (batiskafff) [inline src/types/fpti.js:60]: Isn't it mandatory field?

File: test/lib/fpti-test-utils.js [hunk 3]
```diff
@@ -0,0 +1,74 @@
...
+
+export function verifyFptiIdentityMappings(fptiEvent, identityData) {
+  expect(fptiEvent.confidenceScore).toEqual(identityData.confidenceScore);
+  expect(fptiEvent.encryptedAccountNumber).toEqual(
+    identityData.encryptedAccountNumber
+  );
+  expect(fptiEvent.identificationType).toEqual(identityData.identificationType);
+}
+
+export function verifyFptiDeviceInfoMappings(fptiEvent, deviceInfoInput) {
```
Comment (batiskafff) [inline test/lib/fpti-test-utils.js:57]: Why didn't you use following assert?
```
expect(fptiEvent).toEqual(deviceInfoInput);
```

File: src/lib/shopping-fpti/event-handlers.js (hunk 1 above)
Comment (dturgumbaev) [reply src/lib/shopping-fpti/event-handlers.js:18]: it return new FPTI attributes based on the event type

File: src/types/fpti.js (hunk 2 above)
Comment (dturgumbaev) [reply src/types/fpti.js:60]: nope it is not

File: test/lib/fpti-test-utils.js (hunk 3 above)
Comment (dturgumbaev) [reply test/lib/fpti-test-utils.js:57]: only the subset of fptiEvent attributes are mapped from deviceInfoInput

**Review by batiskafff — APPROVED** [review]

File: src/lib/shopping-fpti/event-handlers.js [hunk 4]
```diff
@@ -0,0 +1,104 @@
+/* @flow */
+import type { Config } from '../../types';
+import type { EventType } from '../../types/shopping-events';
+
+export function eventSifnoBuilderInit(config : Config) : Object {
```
Comment (zacharyhill) [inline src/lib/shopping-fpti/event-handlers.js:5]: Can you fix this to be `eventSinfoBuilderInit` -- looks like `fn` `should be `nf`. Not a big deal but might avoid some confusion in the future.

**Review by zacharyhill — APPROVED** [review]

File: src/lib/shopping-fpti/event-handlers.js (hunk 4 above)
Comment (dturgumbaev) [reply src/lib/shopping-fpti/event-handlers.js:5]: I'll fix it before merge

**Pull request merged**